Method of cleaning up error messages

thijsc opened the pull request:
Add a protected method that we can use to clean up user data in error messages if we encounter common errors that leak data.

luismiramirez commented:
I like the feature

- Do we have a list of libraries/frameworks that write sensitive information in the error text?
- How would the config option for the user look?

thijsc commented:
> Do we have a list of libraries/frameworks that write sensitive information in the error text?

This PG error is the only one I'm aware of at the moment. If we learn of new ones we could add them.

> How would the config option for the user look?

We could start by making this only accessible via Ruby. I guess it could also be a general configuration option.

tombruijn commented:
I think we wanted to research this a bit more in [this direction with a global block](https://github.com/appsignal/integration-guide/issues/77#issuecomment-960541821) that allows users to customize whatever they want to on the error/span/transaction before it being sent. The user who requested this initially also wanted to filter out some more metadata we set by default.

thijsc commented:
Is that research a reason to not ship this approach?

tombruijn commented:
Well it's another thing to deprecate later if we change it.
And we just merged [the process](https://github.com/appsignal/integration-guide/blob/48ae4ffadae1d5c041d35e3b63b84c009a275420/decisions.md) about research first and implement second. So I'd rather follow that now.
If this needs higher priority we can pick it up on Monday.

thijsc commented:
I've reverted this back to the first thing I did, just have a hard-coded helper to fix the most common cases. We could either wrap the cleaner I added into a new hooks system in the future, or keep it as is.

Does that alleviate your concern @tombruijn?

tombruijn commented:
Yes, that fixes the immediate issue.
